boa/src/property/mod.rs

```diff
@@ -293,6 +293,17 @@ pub enum PropertyKey {
     Index(u32),
 }
 
+impl From<&Value> for PropertyKey {
+    #[inline]
+    fn from(value: &Value) -> PropertyKey {
+        match value {
+            Value::String(ref string) => string.clone().into(),
+            Value::Symbol(ref symbol) => symbol.clone().into(),
+            _ => panic!("Cannot be used as property"),
+        }
+    }
+}
```

Comment:
We don't need this we have a spec compliant (ToPropertyKey) `Value::to_property_key` method that handles all the invariants of `Value`.